bindings/rust/s2n-tls-tokio/tests/handshake.rs

```diff
@@ -163,26 +163,18 @@ async fn handshake_error_with_blinding() -> Result<(), Box<dyn std::error::Error
 
     let client = TlsConnector::new(client_config.clone());
```

Comment:
Shouldn't blinding only happen once per connection? Your PR description seems to imply that blinding can occur twice in a row?

Reply:
In the tokio bindings, it can happen twice: https://github.com/aws/s2n-tls/blob/main/bindings/rust/s2n-tls-tokio/src/lib.rs#L407-L415 That's because the tokio bindings don't just shutdown s2n, they also shutdown the underlying tcp stream. So after waiting for the initial blinding, if s2n_shutdown fails in a way that triggers blinding, we apply that blinding before closing the underlying stream.

Reply:
It's kind of hard to follow that part of the tokio bindings. I'm going to clean it up as a follow-up to this. I just don't want to rewrite both the tests and the logic at the same time :)

Reply:
https://github.com/aws/s2n-tls/pull/4358

Reply:
But like, do we want blinding to happen twice? From my understanding that shouldn't be necessary. Is there some reason why we have this behavior?

Reply:
Okay, ignore that "twice" part. Imagine that s2n_shutdown is reading records looking for a close_notify alert. It fails to decrypt a record, which is a blinded error because there have been previous timing attacks related to decrypting records. If we then immediately close the underlying stream, we're providing an immediate error signal without applying the required blinding to defend against timing attacks.